pkg/buildah/imageCrdBuilder.go

```diff
@@ -0,0 +1,310 @@
+						return err
+					}
+					icb.Content.ActionCMD[f.Name()] = string(c)
+				}
+			}
+		}
+	}
+	err = icb.TemplateCmdParse()
+	if err != nil {
+		return err
+	}
+
+	return nil
+}
+
+func (icb *ImageCRDBuilder) AppContentApply() error {
+	utd := unstructured.Unstructured{}
+	utd.Object, _ = runtime.DefaultUnstructuredConverter.ToUnstructured(icb.Content.AppConfig)
+	gvk := icb.Content.AppConfig.GroupVersionKind()
+	dyclient, _ := GetGVRdyClient(&gvk, "default")
+	if _, err := dyclient.Create(context.TODO(), &utd, metav1.CreateOptions{}); err != nil {
+		if utd.GetResourceVersion() == "" {
+			var objGET *unstructured.Unstructured
+			objGET, err = dyclient.Get(context.TODO(), utd.GetName(), metav1.GetOptions{})
+			if err != nil {
+				return errors.Wrap(err, "unable to get obj")
+			}
+			fmt.Println(objGET.GetResourceVersion())
+			utd.SetResourceVersion(objGET.GetResourceVersion())
+		}
+		fmt.Println("dyclient.Create err")
+		if !apierrors.IsAlreadyExists(err) {
+			return errors.Wrap(err, "unable to create secret")
+		}
+		if _, err = dyclient.Update(context.TODO(), &utd, metav1.UpdateOptions{}); err != nil {
+			return errors.Wrap(err, "unable to update secret")
+		}
+	}
+	return nil
+}
+
+func (icb *ImageCRDBuilder) DeleteContainer() error {
+	realImpl, err := New("")
+	if err != nil {
+		return err
+	}
+	return realImpl.Delete(icb.imagename)
+}
+
+func FileReadUtil(filePath string) ([]byte, error) {
+	file, err := os.Open(filePath)
+	if err != nil {
+		fmt.Println(err)
+		return nil, err
+	}
+	defer file.Close()
+	fileinfo, err := file.Stat()
+	if err != nil {
+		fmt.Println(err)
+		return nil, err
+	}
+	filesize := fileinfo.Size()
+	buffer := make([]byte, filesize)
+	_, err = file.Read(buffer)
+	if err != nil {
+		fmt.Println(err)
+		return nil, err
+	}
+	return buffer, err
+}
+
+func (icb *ImageCRDBuilder) ConfigParse(c []byte) error {
+	config := &imagev1.Image{}
+	_, _, err := yaml.NewDecodingSerializer(unstructured.UnstructuredJSONScheme).Decode(c, nil, config)
+	if err != nil {
+		panic(fmt.Errorf("failed to get GVK: %v", err))
+	}
+	icb.Content.AppConfig = config
+	return nil
+}
+
+func (icb *ImageCRDBuilder) TemplateCmdParse() error {
+	for k, v := range icb.Content.ActionTemplate {
+		if icb.Content.AppConfig.Spec.DetailInfo.AppActions.Actions == nil {
+			icb.Content.AppConfig.Spec.DetailInfo.AppActions.Actions = make(map[imagev1.ActionName]imagev1.Template)
+		}
+		icb.Content.AppConfig.Spec.DetailInfo.AppActions.Actions[imagev1.ActionName(k)] = imagev1.Template(v)
+	}
+	for k, v := range icb.Content.ActionCMD {
+		if icb.Content.AppConfig.Spec.DetailInfo.AppActions.CMD == nil {
+			icb.Content.AppConfig.Spec.DetailInfo.AppActions.CMD = make(map[imagev1.ActionName]imagev1.CMD)
+		}
+		icb.Content.AppConfig.Spec.DetailInfo.AppActions.CMD[imagev1.ActionName(k)] = imagev1.CMD(v)
+	}
+	return nil
+}
+
+func GetGVRdyClient(gvk *schema.GroupVersionKind, namespace string) (dr dynamic.ResourceInterface, err error) {
```

Comment:
use this pkg "github.com/labring/sealos/pkg/client-go/kubernetes"?

Reply:
cc https://github.com/labring/sealos/blob/052b39d9d28eb4d0467a0e553c3975e2b963f4a7/pkg/auth/utils/config.go#L44

Reply:
Got it thx